BPR.Mediator/Services/ResultService.cs

```diff
@@ -0,0 +1,114 @@
+using AutoMapper;
+using BPR.Analysis.Mappers;
+using BPR.Analysis.Models;
+using BPR.Analysis.Services;
+using BPR.Mediator.Enums;
+using BPR.Mediator.Models;
+using BPR.Mediator.Validators;
+using BPR.Persistence.Models;
+using BPR.Persistence.Repositories;
+using BPR.Persistence.Utils;
+using Microsoft.Extensions.Logging;
+
+namespace BPR.Mediator.Services;
+
+public class ResultService : IResultService
+{
+    private readonly IResultRepository _resultRepository;
+    private readonly IAnalysisService _analysisService;
+    private readonly IValidatorService _validatorService;
+    private readonly IMapper _mapper;
+    private readonly ILogger<ResultService> _logger;
+    
+    public ResultService(IResultRepository resultRepository, IAnalysisService analysisService, IValidatorService validatorService, IMapper mapper, ILogger<ResultService> logger)
```

Comment:
you're adding logger but it doesn't seem to be used anywhere. We should add logging to all operations including successes